src/SmoFacade/Database.cs

```diff
@@ -15,10 +15,35 @@ public class Database
     private readonly Microsoft.SqlServer.Management.Smo.Database _database;
     private readonly Server _server;
 
+    public Database(DatabaseConfig databaseConfig)
+    { 
+        string connectionString = databaseConfig.ConnectionString;
+        string dbName = databaseConfig.DatabaseName;
+
+        var sqlConnection = new System.Data.SqlClient.SqlConnection(connectionString);
+        var sqlServerConnection = new Microsoft.SqlServer.Management.Common.ServerConnection(sqlConnection);
+        var sqlSmoServer = new Microsoft.SqlServer.Management.Smo.Server(sqlServerConnection);
+        _database = new Microsoft.SqlServer.Management.Smo.Database(sqlSmoServer, dbName); 
+
+        _server = new Server(connectionString);
+        RefreshAndOpenSqlConnection();
+    }
+
     internal Database(Microsoft.SqlServer.Management.Smo.Database database, Server server)
     {
       _database = database;
       _server = server;
+      RefreshAndOpenSqlConnection();
+    }
+
+    ~Database()
+    {
+      _server.SqlConnection.Close();
+    }

```

Comment:
`Dispose` is preferred to descriptor/finalizer in C#, which is garbage collected unlike C++:
https://stackoverflow.com/questions/4898733/when-should-i-create-a-destructor

Reply:
Using the [iDisposable pattern](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose) like Alex is suggesting is what's needed here if the database class needs to manage the sql connection lifetime. That's going to change how the object is used and managed by consumers as well as throughout the codebase.

Perhaps there's another way to allow consumers to manage the lifetime of the connection separate from the lifetime of the database object so we don't need to make this disposable though?

Reply:
Oh wow didn't know that there was a big difference between c#/c++ destructor/finalizer
Had a look at some blogs and it looks like there are many caveats in how c# destructors are handled - which makes it very difficult to write a correct one.

Sure will transition to using the disposable pattern in c# from now on

Reply:
I came up with an alternative implementation to fix the issue - it won't need this level of resource handling anymore, so looks to be a better solution.
What do you guys think?

(changes in most recent commit - `6aaf59c`)

Reply:
^^ scratch the above
I realised this is a long winded way of doing it - consumer can just do the below instead to achieve the same
`Database db = new Server(connString).Database(dbName)`